### PR TITLE
Adds loading prop

### PR DESCRIPTION
**Pull request by alex-mcleod**
When the loading prop is true, an activity indicator component will be rendered. This indicator can be styled and colored using the styleLoading and loadingColor respectively.

### CODE REVIEW

File: Button.js [hunk 1]
```diff
@@ -49,6 +53,12 @@ var Button = React.createClass({
       disabled ? this.props.styleDisabled : null,
     ];
 
+    var children;
```
Comment (tigershen23) [inline Button.js]: Why are you declaring this here when it seems to get overridden on line 62?

Comment (alex-mcleod) [inline Button.js]: Thanks for pointing that out, have removed it now.

File: Button.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ var {
   Text,
   TouchableOpacity,
   View,
+  ActivityIndicatorIOS
```
Comment (andrewljohnson) [inline Button.js:10]: I imagine this library should be cross-platform (iOS/Android).

**Comment by andrewljohnson** [comment]
I'm not the author of this lib, but I'm not sure this would need to be part of the base library. The UIButton library on iOS has no built-in capability... maybe just subclass react-native-button in the project?

**Comment by ide** [comment]
I agree with @andrewljohnson's assessment -- better to compose components instead of growing their surface area when it's not too hard to do so.

**Comment by alex-mcleod** [comment]
Thanks for the feedback @andrewljohnson and @ide, I think that makes sense, I'll close this.

**Pull request closed without merging**